front/lib/swr/connectors.ts

```diff
@@ -50,15 +50,18 @@ export function useConnectorPermissions<IncludeParents extends boolean>({
     GetDataSourcePermissionsResponseBody<IncludeParents>
   > = fetcher;
 
-  let url = `/api/w/${owner.sId}/data_sources/${dataSource.sId}/managed/permissions?viewType=${viewType}`;
-  if (parentId) {
-    url += `&parentId=${parentId}`;
-  }
-  if (filterPermission) {
-    url += `&filterPermission=${filterPermission}`;
-  }
-  if (includeParents) {
-    url += "&includeParents=true";
+  let url = "";
+  if (dataSource) {
```

Comment:
Shouldn't we set disabled if dataSource is null ? what would happen by calling useSWRWithDefaults with empty url ?

Reply:
Yeah you're right, I was setting the disable parameter on the function call, but let's move the logic within the hook 👍🏼